src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/LegacyPackageReferenceProjectProvider.cs

```diff
@@ -80,11 +80,14 @@ public sealed class LegacyPackageReferenceProjectProvider : INuGetProjectProvide
                 return null;
             }
 
+            var targetFramework = await vsProjectAdapter.GetTargetFrameworkAsync();
```

Comment:
Legacy projects can be retargeted. Can you please ensure that the target framework gets updated?

Reply:
Looks like this just works. When I change the target framework for a project, a new LegacyPackageReferenceProject object gets created with the new target framework.

Reply:
Awesome, resolving this.

Reply:
Added a comment.